### PR TITLE
Stop building with `-Werror`

### PR DESCRIPTION
**Pull request by nalimilan**
`-Werror` should not be used in default build flags as it makes the compilation fail every time somebody tries to build with a new version of the compiler which triggers a new warning. It should only be set when running CI.

I saw this when trying to build the RPM for Julia 1.8.0-rc3 on Fedora rawhide (with GCC 12.1.1):
```
threading.c: In function 'lbt_register_thread_interface':
threading.c:44:5: error: 'strcpy' writing one too many bytes into a region of a size that depends on 'strlen' [-Werror=stringop-overflow=]
   44 |     strcpy(getter_names[idx], getter);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
threading.c:43:34: note: destination object of size [0, 9223372036854775805] allocated by 'malloc'
   43 |     getter_names[idx] = (char *) malloc(strlen(getter));
      |                                  ^~~~~~~~~~~~~~~~~~~~~~
threading.c:46:5: error: 'strcpy' writing one too many bytes into a region of a size that depends on 'strlen' [-Werror=stringop-overflow=]
   46 |     strcpy(setter_names[idx], setter);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
threading.c:45:34: note: destination object of size [0, 9223372036854775805] allocated by 'malloc'
   45 |     setter_names[idx] = (char *) malloc(strlen(setter));
      |                                  ^~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:37: build/threading.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/builddir/build/BUILD/julia-1.8.0-beta3/deps/scratch/blastrampoline-d32042273719672c6669f6442a0be5605d434b70/src'
make: *** [/builddir/build/BUILD/julia-1.8.0-beta3/deps/blastrampoline.mk:14: scratch/blastrampoline-d32042273719672c6669f6442a0be5605d434b70/build-compiled] Error 2
make: Leaving directory '/builddir/build/BUILD/julia-1.8.0-beta3/deps'
```
The warning probably deserves some attention but the problem isn't new I guess so there's no reason to worry the particular user who happened to build with a new compiler version.

### CODE REVIEW

**Comment by ViralBShah** [comment]
Merge?

**Comment by staticfloat** [comment]
We should add this back on in CI as a part of this PR then.

**Comment by nalimilan** [comment]
I've added a commit which seems like it should have done that, but the logs show that the compiler was called without `-Werror`. Any ideas?

**Comment by giordano** [comment]
I think flags are overridden at https://github.com/JuliaLinearAlgebra/libblastrampoline/blob/b829b0e1583da58bc80acbcc0cfb83b482530b92/test/utils.jl#L64-L68

**Comment by nalimilan** [comment]
Thanks, seems to work now!

File: test/utils.jl [hunk 1]
```diff
@@ -64,8 +64,8 @@ function build_libblastrampoline()
     cflags_add = needs_m32() ? "-m32" : ""
```
Comment (giordano) [inline test/utils.jl:64]:
```suggestion
    cflags_add = "-Werror" * needs_m32() ? " -m32" : ""
```
so that you don't need to add `-Werror` twice below

**Review by giordano — APPROVED** [review]
Looks good to me, but I think the setting of cflags can be simplified in tests

**Comment by nalimilan** [comment]
Should be good now.

**Comment by giordano** [comment]
Side note, I think the warning was fixed by #71, right?

**Comment by staticfloat** [comment]
Thanks, @nalimilan!

**Pull request merged**